react/components/SKUSelector/index.js

```diff
@@ -1,90 +1,72 @@
 
-import VTEXClasses from './constants/CustomClasses'
-import SelectorItem from './components/SelectorItem'
+import SKUSelector from './components/SKUSelector';
+import { SKUSelectorContainerPropTypes } from './utils/proptypes'
+import { getMainVariationName, getVariationOptions, getMaxSkuPrice, parseSku } from './utils'
 
 import './global.css'
 
 /**
- * Display a list of SKU items of a problem and its specifications.
+ * Parse the skus to retrieve the given varitations.
  */
-export default class SKUSelector extends Component {
-  static propTypes = {
-    /** Product's slug */
-    productSlug: PropTypes.string,
-    /** SKU selected */
-    skuSelected: PropTypes.object,
-    /** Title which describes the SKU Selector Type */
-    title: PropTypes.string.isRequired,
-    /** List of SKU Items */
-    skuItems: PropTypes.arrayOf(PropTypes.shape({
-      /** Name of the SKU Item */
-      name: PropTypes.string.isRequired,
-      /** Images of the SKU item */
-      images: PropTypes.arrayOf(PropTypes.shape({
-        /** URL of source Image */
-        imageUrl: PropTypes.string.isRequired,
-        /** Brief description of the image */
-        imageLabel: PropTypes.string,
-      })).isRequired,
-    })).isRequired,
-  }
+export default (props) => {
+
+  if (!props.skuSelected || !props.skuSelected.variations || props.skuSelected.variations.length === 0) return null
+
+  const skuSelected = props.skuSelected && parseSku(props.skuSelected) || null
+  const skuItems = props.skuItems && props.skuItems.map(sku => parseSku(sku)) || null
+
+  return <SKUSelectorContainer {...props} skuSelected={skuSelected} skuItems={skuItems} />
+}
 
-  static defaultProps = {
-    title: '',
-    skuItems: [],
+/**
+ * Display a list of SKU items of a product and its specifications.
+ */
+class SKUSelectorContainer extends Component {
+  handleSelectSku = (skuId) => {
+    const slug = this.props.productSlug
+
+    this.context.navigate({
+      page: 'store/product',
+      params: { slug },
+      query: `skuId=${skuId}`,
+    })
   }
 
-  getMaxSkuPrice = items => {
-    let maxPrice = 0
-    if (items) {
-      items.forEach(item => {
-        const [{ commertialOffer: { Price } }] = item.sellers
-        maxPrice = Math.max(maxPrice, Price)
-      })
+  render() {
+    const { skuItems, skuSelected: { variations, itemId } } = this.props
+
+    const name = getMainVariationName(variations)
+    const mainVariation = {
+      name,
+      value: this.props.skuSelected[name],
+      options: getVariationOptions(name, skuItems)
     }
-    return maxPrice
-  }
 
-  stripUrl = url => url.replace(/^https?:/, '')
+    const maxSkuPrice = getMaxSkuPrice(skuItems)
+    const secondaryVariation = {}
+
+    const filteredSkus = skuItems.filter(sku => sku[mainVariation.name] === mainVariation.value)
+
+    if (variations.length > 1) {
+      secondaryVariation.name = variations.filter(variation => variation != mainVariation.name)[0]
```

Comment:
Use `!==` operator to compare type and value.

Reply:
👍